store/state_changes.go

```diff
@@ -35,6 +36,18 @@ const (
 func (store *Store) RegisterFileUpload(ctx context.Context, metaData files.StoredRegisteredMetaData) error {
 	logdata := log.Data{"path": metaData.Path}
 
+	// don't register file upload if it is already registered
+	m := files.StoredRegisteredMetaData{}
+	errFindingMetadata := store.metadataCollection.FindOne(ctx, bson.M{fieldPath: metaData.Path}, &m)
```

Comment:
We also need to check if it's in the same collection as previously too, so add a search criteria on collection id.  I ran a test where I approved a collection and then took it out of the queue.  Because my local setup kept erroring when I tried to delete the file so I could update it, I deleted the collection and started again.  While the file metadata updated, it was still listed as in the old collection so it got confused and didn't publish the files.